apps/address-service/domains/common/utils/services/search/AbstractSearchProvider.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @typedef {Object} NormalizedSearchData
+ * @property {string} country "Russia"
+ * @property {string?} region "Sverdlovsk region"
+ * @property {string} city "Yekaterinburg"
+ * @property {string?} street "Lenina"
+ * @property {string} building "66", "66a"
+ * @property {string?} unitType "room", "flat", "box"
+ * @property {string?} unitName "428", "42/8"
+ */
+
+/**
+ * @typedef {Object} NormalizedSearch
+ * @property {string} value "Russia, Sverdlovsk region, Yekaterinburg, Lenina, 66"
+ * @property {NormalizedSearchData} data
+ */
+
+const { searchContexts } = require('@address-service/domains/common/constants/contexts')
+const get = require('lodash/get')
+
+/**
+ * @abstract
+ */
+class AbstractSearchProvider {
+
+    /**
+     * @abstract
+     * @public
+     * @returns {string} The provider name (constant)
+     */
+    getProviderName () {
+        throw new Error('Method still not implemented.')
+    }
+
+    /**
+     * Returns the context object
+     * @param {?string} context
+     * @returns {Object}
+     * @protected
+     */
+    getContext (context = null) {
+        return {
+            ...get(searchContexts, ['default', this.getProviderName()], {}),
+            ...(
+                context
```

Comment:
May this be replaced with only lodash get expression?
Like `get(searchContexts, [context, this.getProviderName()], {})`

Reply:
Yes, looks good. Refactored.